File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03503/p03503_num4_parsed.py
import itertools

def calculate_profit(N, F, P):
    max_profit = float('-inf')
    for open_periods in itertools.product([0, 1], repeat=10):
        profit = 0
        for i in range(N):
            count = 0
            for j in range(5):
                for k in range(2):
                    if open_periods[j*2+k] and F[i][j][k]:
                        count += 1
            profit += P[i][count]
        max_profit = max(max_profit, profit)
    return max_profit

N = int(input())
F = []
for _ in range(N):
    F.append([list(map(int, input().split())) for _ in range(5)])
P = [list(map(int, input().split())) for _ in range(N)]

result = calculate_profit(N, F, P)
print(result)

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_cpp/p03503/p03503_num4_parsed.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    int N;
    cin >> N;
    
    vector<vector<vector<int>>> F(N, vector<vector<int>>(5, vector<int>(2)));
    for (int i = 0; i < N; i++) {
        for (int j = 0; j < 5; j++) {
            for (int k = 0; k < 2; k++) {
                cin >> F[i][j][k];
            }
        }
    }
    
    vector<vector<int>> P(N, vector<int>(11));
    for (int i = 0; i < N; i++) {
        for (int j = 0; j <= 10; j++) {
            cin >> P[i][j];
        }
    }
    
    int max_profit = -1e9;
    
    for (int mask = 1; mask < (1 << 10); mask++) {
        vector<int> open_periods(N);
        for (int i = 0; i < N; i++) {
            for (int j = 0; j < 5; j++) {
                for (int k = 0; k < 2; k++) {
                    if (F[i][j][k] && ((1 << (2 * j + k)) & mask)) {
                        open_periods[i]++;
                    }
                }
            }
        }
        
        int profit = 0;
        for (int i = 0; i < N; i++) {
            profit += P[i][open_periods[i]];
        }
        
        max_profit = max(max_profit, profit);
    }
    
    cout << max_profit << endl;
    
    return 0;
}